### DISCUSSION TITLE
Is jgroup 5.x.x.x supported on openJDK 17?

### DISCUSSION THREAD

**Discussion by 4devwithgit** [belaban/JGroups#701] [Q&A] [answered]
Is jgroup 5.x.x.x supported on openJDK 17?
We are using jgroups, 4.2.10, with IBM JDK.
We are planning to move to openJDK 17.

Also, when is EOS for Jgroups 4.x.x.x

**Comment by rhusar** [comment, marked as answer]
> Is jgroup 5.x.x.x supported on openJDK 17?

Yes.

Our CI tests compilation on all LTS versions, laest GA version and latest preview version: https://github.com/belaban/JGroups/blob/master/.github/workflows/ci.yml#L18

**Comment by belaban** [comment]
Yes, absolutely! The advantage of running on JDK 17 (or higher) is that you can use virtual threads (`virtual_threads="true"`). You have to use JGroups 5.x though: 4.x does not do this, although it will run on JDK 17+, too.

**Comment by 4devwithgit** [comment]
Also, when is EOS for Jgroups 4.x.x.x ?

**Comment by belaban** [comment]
JGroups 4.x is still used by Infinispan, so no date planned for an EOD. If not used inside a product, it won't be supported. I do community support for 5.x, and if a critical bug is encountered, also for 4.x.Hope this helps, cheersBela Ban-------- Original message --------From: 4devwithgit ***@***.***>Date: Wed, 1 Mar 2023, 12:09To: belaban/JGroups ***@***.***>Cc: Bela Ban ***@***.***>, Comment ***@***.***>Subject: Re: [belaban/JGroups] Is jgroup 5.x.x.x supported on openJDK 17? (Discussion #701)
Also, when is EOS for Jgroups 4.x.x.x ?
